diff --git a/gcc/gcc-4.2.1/gcc/config/arm/unwind-arm.c b/gcc/gcc-4.2.1/gcc/config/arm/unwind-arm.c
index 41d89d1..976901e 100644
--- a/gcc/gcc-4.2.1/gcc/config/arm/unwind-arm.c
+++ b/gcc/gcc-4.2.1/gcc/config/arm/unwind-arm.c
@@ -43,7 +43,7 @@ bool __attribute__((weak)) __cxa_type_match(_Unwind_Control_Block *ucbp,
 					    const type_info *rttip,
 					    void **matched_object);
 
-_Unwind_Ptr __attribute__((weak))
+_Unwind_Ptr
 __gnu_Unwind_Find_exidx (_Unwind_Ptr, int *);
 
 /* Misc constants.  */
diff --git a/gcc/gcc-4.4.0/gcc/config/arm/unwind-arm.c b/gcc/gcc-4.4.0/gcc/config/arm/unwind-arm.c
index 4eb1821..4e92149 100644
--- a/gcc/gcc-4.4.0/gcc/config/arm/unwind-arm.c
+++ b/gcc/gcc-4.4.0/gcc/config/arm/unwind-arm.c
@@ -40,7 +40,7 @@ bool __attribute__((weak)) __cxa_type_match(_Unwind_Control_Block *ucbp,
 					    bool is_reference,
 					    void **matched_object);
 
-_Unwind_Ptr __attribute__((weak))
+_Unwind_Ptr
 __gnu_Unwind_Find_exidx (_Unwind_Ptr, int *);
 
 /* Misc constants.  */
